### ISSUE TITLE
docs: add documentation for working with wasm-pack outside of template

### ISSUE THREAD

**Issue by lrlna** [rustwasm/wasm-pack#533]
## 💡 Feature description
Although the template is a helpful resource to get started with `wasm-pack`, I find that documentation for working with it in more complex situation is lacking.

I am hoping that people who are also working outside of the provided template might jump in with some of the things they've had to tweak, so we can gather a wider range of use cases \o/

I've gathered a few things that I've ran into. Hopefully at least some of this information is useful for putting together more extensive documentation. 

### Importing async
To be able to add `.wasm` to the project, it needs to be imported async. Templates use a wrapper file to wrap the entire application in an async `import`. Subsequently, you can do a standard import of your webpack module. @ashleygwilliams helped me out figuring this out with a drawing, I made v2.0 of the said drawing:
![module-loading](https://user-images.githubusercontent.com/8107784/51910336-340d4c80-23cf-11e9-9cdc-1ab2e2316384.png)

This approach works if you are able to change the way your application builds. For those who are _not_ able to change to their build, I solved the problem with just a dynamic import of the my module:

1. Dynamic import does not work everywhere ([can i use](https://caniuse.com/#feat=es6-module-dynamic-import)), so one will very likely require a babel plugin to handle this. [babel-plugin-syntax-dynamic-import](https://www.npmjs.com/package/babel-plugin-syntax-dynamic-import) is what i used:

      ```js
      // in .babelrc
      {  
        "plugins": [
          "syntax-dynamic-import"
        ]
      }
      ```

      Otherwise you end up with this error:
      <img width="546" alt="screen shot 2019-01-29 at 13 12 42" src="https://user-images.githubusercontent.com/8107784/51907599-96fae580-23c7-11e9-9cd3-4a296ebf8d0c.png">
2. Then inside the project something like this can be done:
      ```js
      // initial import that will have to be used as a promise later on in code
      const schemaWasm = import('mongodb-schema-parser');

      function getSchema(doc) { 
        // .then on the previously imported module
        schemaWasm
          .then(module => {
            // use the module's API
            const schemaParser = new module.SchemaParser();
            schemaParser.write(doc)
            return schemaParser.toJson();
          })
          .catch(e => return new Error(`Error in mongodb-schema-parser ${e}`))
      }
      ```

### Webpack configuration
Depending on people's webpack version and build target a few options are possible:
- build target that's _not_ `electron-renderer` can use webpack < 4.x, but a few things might need to be added:
  1. if default [resolve extensions](https://webpack.js.org/configuration/resolve/#resolve-extensions) were overwritten by base config, might have to add `.wasm` to that array. For example:
        ```js
        extensions: ['.js', '.jsx', '.json', 'less', '.wasm']
        ```
      Otherwise you get errors like these:
      <img width="724" alt="screen shot 2019-01-29 at 12 11 40" src="https://user-images.githubusercontent.com/8107784/51904493-13d59180-23bf-11e9-9d76-7128e4388317.png">

  2. _i think_ (but might be wrong) you might need to add a wasm loader plugin. I've used [this one](https://github.com/ballercat/wasm-loader) when experimenting and it worked quite well. It just needs to be added to [module rules](https://webpack.js.org/configuration/module/#module-rules): 
        ```js
        module: {
          rules: [{
            test: /\.wasm$/,
            include: [/node_modules/],
            use: [ { loader: 'wasm-loader'} ]
          }]
        }
        ```
        Otherwise you get a bunch of these errors:
          <img width="654" alt="screen shot 2019-01-22 at 18 33 21" src="https://user-images.githubusercontent.com/8107784/51904353-b93c3580-23be-11e9-9d25-8e20f0df880d.png">

- if people are building electron (like me), [webpack did not support](https://github.com/webpack/webpack/issues/7677) wasm with electron target so people will need to upgrade to webpack > 4.16.0. Likely nothing will have to be done to the loader, but depending on whether `resolve extensions` have been overwritten, `.wasm` might need to be added to that.

**Comment by ashleygwilliams** [comment]
@lrlna would you be interested in adding some of this documentation? I'd welcome PRs and would be happy to help you get started if you'd like! Let me know!

**Comment by lrlna** [comment]
hey @ashleygwilliams yea I can add some of this in sometime in the next two weeks. If anyone has different experiences, we can always update I suppose! Where does the book's source live exactly? And what section would you like me to add this to?

**Comment by MaxBittker** [comment]
Thanks for writing such a detailed issue! I'm having the issue you describe can result from not having
`@babel/plugin-syntax-dynamic-import` ... but i think I do have this plugin. It's possible I've just made other webpack errors, but posting this in case others are experiencing the problem

**Comment by MaxBittker** [comment]
oh... there's a bug in webpack 4.29.0
https://github.com/webpack/webpack/issues/8656

**Comment by lrlna** [comment]
@MaxBittker, yep there was a bug last week! Try and see if  `4.29.6` works for you (it worked for me).

**Comment by lirhtc** [comment]
Hi @lrlna, thanks for the drawing. It makes the whole import process very clear. I am now trying to figure out how to import a wasm module in an Electron app and have no idea to do so. Is that possible to provide some more information about how to do it please?

**Comment by lrlna** [comment]
hey @lirhtc! if you have webpack > 4.16.0, you should be able to import it with wherever you need it. If you have babel running, I'd also add this [babel-dynamic-import-plugin](https://www.npmjs.com/package/babel-plugin-syntax-dynamic-import). I think the first two steps I've outlined in `Importing Async` + having webpack > 4.16.0 _should_ work. Let me know if it doesn't! Happy to help brainstorm!

```js
var wasm = import('wasm')
function runWasmAction (param) { 
  // .then on the previously imported module
  wasm
    .then(module => {
      // use your module's API
    })
    .catch(e => return new Error(`Error in wasm module ${e}`))
}
```
